spec/functions/number.md

```diff
@@ -257,8 +257,7 @@ Using this _option_ in such a case results in a _Bad Option_ error.
 The value of the _operand_'s `currency` MUST be either a string containing a
 well-formed [Unicode Currency Identifier](https://unicode.org/reports/tr35/tr35.html#UnicodeCurrencyIdentifier)
 or an implementation-defined currency type.
-Although currency codes are expected to be uppercase,
-implementations SHOULD treat them in a case-insensitive manner.
+Currency codes are case-insensitive.
```

Comment:
This leaves it unclear what the resolved value is. Could we be more explicit?
```suggestion
The resolved `currency` value is normalized to upper-case.
```

Reply:
As a reminder, even ASCII letters have locale-sensitive case folding exceptions (Turkic languages).

Do we need to specify that the resolved value is uppercased? Mostly we treat the operands as immutable. We don't have to require that implementations perform the case fold operation. It's a good idea to do it to save later processing, but we can't really test it at the MF level except to say that a value with `usd` is as functional as one with `USD`.

Reply:
I think we should specify the casing, so that it's clear in a message like
```
.input {$n :currency currency=Usd}
{{{$n :foo:bar}}}
```
what `currency` value the `:foo:bar` function handler will see.

```suggestion
The resolved `currency` value is locale-insensitively normalized to upper-case.
```

Reply:
I have a variety of quibbles with the suggestion, starting with not using the defined term _resolved value_. Also, we allow implementation-defined (currency) types in this option also. Since the literals must be ASCII letters, we can use Unicode's casefold algorithm. Perhaps:

```suggestion
Currency codes are case-insensitive.
If the value of the `currency` _option_ is a _literal_, the _resolved value_ MUST be case-folded to uppercase.
```

Reply:
Why limit the case-folding to literal values only? Why should the resolved value be different for `$a` vs `$b`?
```
.local $cc = {eur}
.input {$a :currency currency=eur}
.input {$b :currency currency=$cc}
...
```

Reply:
The value would have been resolved before reaching the function. So it is either resolved as a literal or as an implementation-defined type. We can't uppercase a `Currency` object, but we can on a literal. Note that your examples are in the MF processor space, but this directive is inside the function handler.

Reply:
We don't currently have any place in the spec that defines specific behaviour based on whether a value originates from a _literal_, and I would very much prefer not adding anything like that. The restriction on `:number select` makes sense and is only about the immediate _option value_ in part to avoid needing to track the origins of variable values like this.

Given that we say that
> The _resolved value_ of a _text_ or a _literal_ contains the character sequence of the _text_ or _literal_ after any character escape has been converted to the escaped character.

The definition here ought to limit its impact not on whether the option's resolved value came from a _literal_, but on whethe or not it contains a character sequence.

Reply:
Why can't we just require that the currency is upper-case, and then we don't need to worry about the case folding stuff.

Reply:
Also, "case-folded to uppercase" doesn't make sense. Unicode Case Mapping defines algorithms for Uppercase, Lowercase, Titlecase, and Case Fold. You don't "case fold to uppercase".